lib/IPParser.js

```diff
@@ -1,44 +1,59 @@
-var ipp = module.exports = function(ip){
-	if(ip.indexOf('.')!==-1){
-		return ipp.parseIPv4(ip);
-	}else{
-		return ipp.parseIPv6(ip);
-	}
+'use strict';
+
+var IPv6 = require('ip-address').v6;
+var IPv4 = require('ip-address').v4;
+
+module.exports = IPParser;
+
+function IPParser(ip) {
+  if (ip.indexOf('.') !== -1) {
+    return IPParser.parseIPv4(ip);
+  } else {
+    return IPParser.parseIPv6(ip);
+  }
+}
+
+IPParser.parseIPv4 = function parseIPv4(ip) {
+  var v4Address = new IPv4.Address(ip)
+  if(!v4Address.isValid()) {
+    throw new Error("Invalid IPv4 address");
+  }
+  return(ipv4Buffer(v4Address.parsedAddress));
 };
 
-ipp.invalidIPv4 = /[^0-9\.]/i
-ipp.parseIPv4 = function(ipstr){
-	if(ipp.invalidIPv4.test(ipstr)){
-		throw new Error("Invalid IPv4 address");
-	}
-	var arr = new Buffer(4);
-	arr.fill(0);
-	ipstr.split('.').forEach(function(nr,i){
-		arr[i] = parseInt(nr);
-	});
-	return arr;
+IPParser.parseIPv6 = function parseIPv6(ip) {
+  var v6Address = new IPv6.Address(ip);
+  if(!v6Address.isValid()) {
+    throw new Error("Invalid IPv6 address");
+  }
+  return(ipv6Buffer(v6Address.parsedAddress));
 };
 
-var repeat = function(c,l){
-	var str="",i=0;
-	while(i++ < l)str+=c;
-	return str;
+
+function ipv4Buffer(groups) {
+  var arr = new Buffer(4);
+  arr.fill(0);
+  groups.forEach(function part(nr, i) {
+    arr[i] = parseInt(nr);
+  });
+  return arr;
+}
+
+function ipv6Buffer(groups) {
+  var arr = new Buffer(16);
+  arr.fill(0);
+  groups.forEach(function part(hex, i) {
+    if (hex == "") return;
+    if (hex.length < 4) {
+      hex = repeat('0', 4 - hex.length) + hex;
+    }
+    arr.write(hex, i * 2, 'hex');
+  });
+  return arr;
+}
+
+function repeat(c, l) {
+  var str = "", i = 0;
+  while (i++ < l)str += c;
+  return str;
 }
-ipp.invalidIPv6 = /[^0-9a-f\:]/i;
-ipp.parseIPv6 = function(ipstr){
-	if(ipp.invalidIPv6.test(ipstr)){
-		throw new Error("Invalid IPv6 address");
-	}
-	var arr = new Buffer(16);
-	arr.fill(0);
-	var l = ipstr.split(':').length;
-	if(l < 8) ipstr = ipstr.replace('::',repeat(':',10 - l));
-	ipstr.split(':').forEach(function(hex,i){
-		if(hex=="")return;
-		if(hex.length < 4){
-			hex = repeat('0',4 - hex.length) + hex;
-		}
-		arr.write(hex,i*2,'hex');
-	});
-	return arr;
-};
```

Comment:
does it make sense to make a wrapper for ip-address package instead of just directly using it?